align_system/algorithms/oracle_adm.py

```diff
@@ -0,0 +1,321 @@
+
+            def _get_treatment(poss_treatments: List[str], injury_name: str, injury_location: str) -> Optional[str]:
+                """
+                Return appropriate treatment for given injury name and location, given available supplies. If no
+                treatment exists under these conditions, None will be returned.
+                """
+                match injury_name:
+                    case 'Amputation':
+                        if 'Tourniquet' in poss_treatments:
+                            return 'Tourniquet'
+                    case 'Burn':
+                        if 'Burn Dressing' in poss_treatments:
+                            return 'Burn Dressing'
+                    case 'Broken Bone':
+                        if 'Splint' in poss_treatments:
+                            return 'Splint'
+                    case 'Chest Collapse':
+                        if 'Decompression Needle' in poss_treatments:
+                            return 'Decompression Needle'
+                    case 'Laceration':
+                        if 'thigh' in injury_location:
+                            if 'Tourniquet' in poss_treatments:
+                                return 'Tourniquet'
+                        else:
+                            if 'Pressure bandage' in poss_treatments:
+                                return 'Pressure bandage'
+                    case 'Puncture':
+                        if 'bicep' in injury_location or 'thigh' in injury_location:
+                            if 'Tourniquet' in poss_treatments:
+                                return 'Tourniquet'
+                        else:
+                            if 'Hemostatic gauze' in poss_treatments:
+                                return 'Hemostatic gauze'
+                    case 'Shrapnel':
+                        if 'face' in injury_location:
+                            if 'Nasopharyngeal airway' in poss_treatments:
+                                return 'Nasopharyngeal airway'
+                        else:
+                            if 'Hemostatic gauze' in poss_treatments:
+                                return 'Hemostatic gauze'
+                    case 'Internal':
+                        return 'Pain Medications'
+                    case _:
+                        raise ValueError(f"Unknown injury type: {injury_name}")
+
+                return None
+
+            # Determine what treatment to apply, narrow down applicable characters
+            if 'treatment' not in action_to_take.parameters:
+                # Can only apply treatments if supplies are available
+                poss_treatments = [s.type for s in scenario_state.supplies if s.quantity > 0]
+                ta3_server_treatments = [
+                    'Burn Dressing', 'Decompression Needle', 'Hemostatic gauze', 'Nasopharyngeal airway',
+                    'Pressure bandage', 'Splint', 'Tourniquet', 'Pain Medications'
+                ]
```

Comment:
Ideally these values would be pulled from an enum, I'm guess it's not in the case due to differences between the client enums and what's supported on the server?